keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,899 @@
+
+### Timeline:
+v1.33 (04/2025): 
+- We announce the removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+- We add the `GitRepoVolumeDriver` feature-gate to kubelet and
+default it to `false`.
+
+v1.34 (08/2025):
+- No change.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+v1.35 (11/2025):
+- No change.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+v1.36 (04/2026):
+- Lock the feature-gate, i.e. users cannot change its value.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+v1.37 (08/2026): 
+- No change.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+v1.38 (11/2026):
+- No change.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+v1.39 (04/2027):
+- We will remove the feature-gate.
+- We will remove the gitRepo driver code.
+- We will announce removal in kubernetes release notes, documentation and via
+kubernetes release blog post.
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
```

Comment:
We do need integration and/or e2e to ensure this one.

Reply:
Yes we do! Added the details about testing.